tokio/src/runtime/scheduler/multi_thread/queue.rs

```diff
@@ -107,12 +107,19 @@ pub(crate) fn local<T: 'static>() -> (Steal<T>, Local<T>) {
 impl<T> Local<T> {
     /// Returns the number of entries in the queue
     pub(crate) fn len(&self) -> usize {
-        self.inner.len() as usize
+        let (_, head) = unpack(self.inner.head.load(Acquire));
+        // safety: this is the **only** thread that updates this cell.
+        let tail = unsafe { self.inner.tail.unsync_load() };
+        len(head, tail)
     }
 
     /// How many tasks can be pushed into the queue
     pub(crate) fn remaining_slots(&self) -> usize {
-        self.inner.remaining_slots()
+        let (steal, _) = unpack(self.inner.head.load(Acquire));
+        // safety: this is the **only** thread that updates this cell.
+        let tail = unsafe { self.inner.tail.unsync_load() };
+
+        LOCAL_QUEUE_CAPACITY - len(steal, tail)
```

Comment:
Why not just `LOCAL_QUEUE_CAPACITY - self.len()`?

Reply:
![pr](https://github.com/user-attachments/assets/286c26ae-4275-44a4-b801-6e6e5c377458)

Thanks for your review, I might be mistaken, but I thought the slots that are being stolen was not available for pushing new task.

Reply:
Oh I see now. I did not notice that it was using `steal` instead of `head` for the first parameter to `len`.